packages/grid/x-data-grid/src/models/gridSlotsComponent.ts

```diff
@@ -10,7 +11,7 @@ export interface GridSlotsComponent extends GridIconSlotsComponent {
    * The custom Checkbox component used in the grid for both header and cells.
    * @default Checkbox
    */
-  BaseCheckbox: React.JSXElementConstructor<any>;
+  BaseCheckbox: React.JSXElementConstructor<NonNullable<GridSlotsComponentsProps['baseCheckbox']>>;
```

Comment:
This change is interesting. Without it, you can pass any component to a slot and TypeScript would accept it:
```tsx
{
  baseCheckbox: (props: { whatever: number }) => {
    return <div>{props.whatever}</div>;
  }
},
```

Not sure if changing it across all of the `GridSlotsComponent` members is worth it right now.

cc @MBilalShafi

Reply:
Oh, and most importantly - the change above prevents us from passing anything to the `rootProps.slots.baseCheckbox` slot.

```tsx
<rootProps.slots.baseCheckbox
 whatever={number} // TypeScript won't complain about this if `baseCheckbox` props are typed as `any`
/>
```

Reply:
Yes, I think it could be a very nice addition, and totally worth it to do. Thank you for identifying this 👍

Reply:
After trying this out, I feel this is an aggressive change and requires some time as many references need to be fixed.
For instance, for Grid components we can not directly use this type `GridSlotsComponentsProps['componentName']>` because it uses `Partial` inside and typescript throws type errors.
When Joy base slots are introduced they might also introduce types mismatch and we'll have to fix them too. So keeping these as-is for now (JSXElementConstructor<any>) won't make that a breaking change later on.

So I propose we do release as planned and do this change in a minor release or if it causes a considerable migration impact, plan it for v7. What do you think?

Reply:
@cherniavskii I think this should be reverted.

Reply:
Done!
We can solve this problem later